src/weapons/GunController.gd

```diff
@@ -0,0 +1,57 @@
+class_name GunController
+extends Node3D
+
+@export var weapon_aim_speed = 0.1
+
+var b := false
+func _process(delta):
+	if Input.is_action_just_pressed("ui_cancel"):
+		b = not b
+		if b:
+			$CameraController.capture_mouse()
+		else:
+			$CameraController.release_mouse()
+	
+	var gun = get_gun()
+	if gun:
+		if Input.is_action_pressed("shoot"):
+			gun.shoot_pressed()
+		if Input.is_action_just_released("shoot"):
+			gun.shoot_released()
+	
+	if Input.is_action_pressed("rmb"):
+		var target = $CameraController.get_camera_centered_target()
+		
+		if target and gun:
+			# Slew the weapon to the specific target at the center of the camera, in the world.
+			
+			var offset = Vector3.ZERO
+			if gun.has_method('get_weapon_offset'):
+				offset = gun.get_weapon_offset()
+			var source_pos = get_gun().global_position + offset
+			$Looker.look_at_from_position(source_pos, target, Vector3.UP)
+			var r : Vector3 = target - $HBase.global_position
+			
+			var forward = (-$HBase.global_basis.z)
+			var forward_vert = (-$HBase/VBase.global_basis.z)
+			
+			var sign_ = sign(r.y)
```

Comment:
Have you worked with C# before? Because having a trailing _ to identify members is something you see in that language :0